File: tutorial/patterns/p23.cpp
#include <iostream>
using namespace std;

int main(){
    int n;
    cin >> n;
    int i = 1;
    int count = 1;

    while (i<= n)
    {
    int j = 1;
    int space = n-i;

    while (space)
    {
        cout << " ";
        space--;
    }
    while (j <= i)
    {
        cout << count;
        j++;
        count++;
    }
    cout << endl;
    i++; 
    }
}

File: tutorial/patterns/p23.dart
import 'dart:io';

void main() {
  int n = int.parse(stdin.readLineSync()!);
  int i = 1;
  int count = 1;

  while (i <= n) {
    int space = n - i;

    int j = 1;
    while (j <= space) {
      stdout.write(" ");
      space--;
    }

    while (j <= i) {
      stdout.write(count);
      count++;
      j++;
    }
    stdout.writeln();
    i++;
  }
}
